lib/ezy_clients.dart

```diff
@@ -34,12 +31,12 @@ class EzyClients {
     clients[client.name] = client;
   }
 
-  EzyClient getClient(String clientName) {
-    var client = clients[clientName]!;
+  EzyClient? getClient(String clientName) {
+    EzyClient? client = clients[clientName];
     return client;
   }
 
-  EzyClient getDefaultClient() {
-    return clients[defaultClientName]!;
+  EzyClient? getDefaultClient() {
+    return clients[defaultClientName];
```

Comment:
Chỗ này không thể thay đổi được vì có thể gây breaking change cho các ứng dụng đang sử dụng, nếu muốn em phải tạo hàm mới để đảm bảo nguyên tắc Open/Close

Reply:
Chỗ này buộc phải thay đổi, nó có nguy cơ khá lớn nếu không tìm thấy client -> dẫn đến việc EzyClient đang được khai báo là non-null nhưng bị gán null vào -> khả năng phát sinh lỗi. 
Chỗ này mình có thể mark deprecated rồi mình thực hiện breaking change cho version sau sẽ ổn hơn ạ.

Reply:
Chỗ này thực ra anh cũng đã từng để `EzyClient?` nhưng khi sử dụng cực kỳ bất tiện, vì trong thực tế sử dụng các client sẽ được khởi tạo và được đưa vào quản lý tương ứng với tên và không bao giờ `null`, khi sử dụng nếu cứ để `?` thì đầu sử dụng cứ phải thêm `!` cũng bất tiện nên anh nghĩ cứ để đó không cần phải `deprecated`, có thể tạo thêm hàm `?` cũng không sao.

Reply:
> Chỗ này thực ra anh cũng đã từng để `EzyClient?` nhưng khi sử dụng cực kỳ bất tiện, vì trong thực tế sử dụng các client sẽ được khởi tạo và được đưa vào quản lý tương ứng với tên và không bao giờ `null`, khi sử dụng nếu cứ để `?` thì đầu sử dụng cứ phải thêm `!` cũng bất tiện nên anh nghĩ cứ để đó không cần phải `deprecated`, có thể tạo thêm hàm `?` cũng không sao.

Vậy chỗ này anh đang chấp nhận đánh đổi rủi ro để lấy sự tiện lợi trong quá trình dev ạ?
Em thấy nó không đủ an toàn á anh, đồng thời các biến nullable đều phải có check `if(x!=null)` trước thì mới được thêm `!`, nếu không nó phải được access bằng `x?.y?.z` thay vì `x!` luôn ạ.

Reply:
Anh nghĩ là để `EzyClient?` cũng là ý tưởng tốt, tuy nhiên phương thức hiện tại nó đã không để là `EzyClient?` rồi giờ sửa trực tiếp sẽ vi phạm nguyên tắc Open/Close không phải là ý hay, anh nghĩ là có thể tạo hàm mới, còn về deprecated thì anh nghĩ cứ để suy nghĩ thêm em ạ.